src/apps/virtio_net/virtio_net.lua

```diff
@@ -0,0 +1,76 @@
+-- Application to connect to a virtio-net driver implementation
+--
+-- Licensed under the Apache 2.0 license
+-- http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Copyright (c) 2015 Virtual Open Systems
```

Comment:
@lukego, regarding the "cargo-cult from each other", here is a reference to the Apache 2.0 as it is included in the SnabbSwitch source code itself https://github.com/SnabbCo/snabbswitch/blob/master/COPYING#L178:

```
APPENDIX: How to apply the Apache License to your work.

   To apply the Apache License to your work, attach the following
   boilerplate notice, with the fields enclosed by brackets "[]"
   replaced with your own identifying information. (Don't include
   the brackets!)  The text should be enclosed in the appropriate
   comment syntax for the file format. We also recommend that a
   file or class name and description of purpose be included on the
   same "printed page" as the copyright notice for easier
   identification within third-party archives.

Copyright [yyyy] [name of copyright owner]

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
```

And if still there are doubts, here is also a similar reference to the GPL licensing How-To  http://www.gnu.org/licenses/gpl-howto.en.html:

```
Whichever license you plan to use, the process involves adding two elements to each source file
 of your program: a copyright notice (such as “Copyright 1999 Terry Jones”), and a statement of
copying permission, saying that the program is distributed under the terms of the GNU General Public License
(or the Lesser GPL).
```

I am not a lawyer, but it appears to me that it is "mandatory" to put those license headers in place, it is not even optional.